pytun_pmd3/wintun.py

```diff
@@ -179,6 +188,10 @@ def raise_last_error() -> None:
     raise Exception(get_error_message(get_last_error()))
 
 
+async def wait_for_event(handle: HANDLE) -> None:
+    await asyncio.get_running_loop().run_in_executor(None, lambda: kernel32.WaitForSingleObject(handle, INFINITE))
```

Comment:
I'm worried canceling here will cause the thread calling `WaitForSingleObject` to keep running.
Maybe something like this could solve it:
```python
try:
    await asyncio.get_running_loop().run_in_executor(None, lambda: kernel32.WaitForSingleObject(handle, INFINITE))
except asyncio.CancelledError:
    kernel32.SetEvent(handle)
    raise
```

Reply:
done